src/traces/scatter/plot.js

```diff
@@ -72,7 +72,7 @@ module.exports = function plot(gd, plotinfo, cdscatter, scatterLayer, transition
             });
         });
     } else {
-        scatterLayer.selectAll('g.trace').each(function(d, i) {
+        join.each(function(d, i) {
```

Comment:
@etpinard it doesn't seem like animations can remove traces, is that correct? That's what it looks like, since I don't see a `join.exit()` associated with them (since `isFullReplot = !transitionOpts`)... yet the comment above implies that you can *add* traces:
```js
// Must run the selection again since otherwise enters/updates get grouped together
// and these get executed out of order. Except we need them in order!
```
So... do I need to worry about animations deleting traces or not?

Reply:
> do I need to worry about animations deleting traces or not?

Right, removing traces with `'frame.redraw': false` (i.e. without a replot at the end of the animation), doesn't work currently:

![peek 2018-09-13 16-17](https://user-images.githubusercontent.com/6675409/45513370-79a6e100-b770-11e8-93f5-233e736d49f0.gif)

https://codepen.io/etpinard/pen/ZMRReB

so I don't think you broke anything here.

Reply:
Ah Ricky wrote an issue about this https://github.com/plotly/plotly.js/issues/932